### PR TITLE
Seed handle

### PR DESCRIPTION
**Pull request by cloudhead**
Allows external code to ask the seed for projects and peers.

### CODE REVIEW

File: seed/src/lib.rs [hunk 1]
```diff
@@ -206,21 +206,60 @@ impl Default for NodeConfig {
     }
 }
 
+pub struct NodeHandle {
+    channel: chan::UnboundedSender<Request>,
+}
+
+impl NodeHandle {
+    pub async fn get_projects(&mut self) -> Result<Vec<()>, chan::SendError> {
```
Comment (xla) [inline seed/src/lib.rs:214]: The one thing I'd change about the proposed API is to not leak this `SendError` as it defeats the purpose of hiding channel semantics from consumers.

Comment (cloudhead) [reply seed/src/lib.rs:214]: That's a :ok_hand: point

Comment (cloudhead) [reply seed/src/lib.rs]: 03fa6bd

**Review by FintanH — COMMENTED** [review]
I'm finding it a bit hard to review this PR. It seems like there's a lot of different types and methods tossed in `lib.rs` at this stage. Could we start creating sub-modules to better organise this?

**Review by FintanH — APPROVED** [review]
SO much easier! LGTM :clap:

File: seed/src/event.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        name: Option<String>,
+    },
+    /// A peer has disconnected.
+    PeerDisconnected(PeerId),
+    /// A project has been tracked from a peer.
+    ProjectTracked(Project, PeerId),
+}
+
+impl Event {
+    pub(crate) async fn peer_connected(
```
Comment (xla) [inline seed/src/event.rs:25]: What's the reason for the excessive use of crate visibility? Why not instead control the re-exports?

**Review by xla — APPROVED** [review]
🕌 ⤵️ 📞 👈

File: seed/src/event.rs (hunk 2 above)
Comment (cloudhead) [reply seed/src/event.rs:41]: Well the thing is I need to export `Event`, but having this as simply `pub` will allow users to call `Event::peer_connected` afaict.

**Pull request merged**